src/ai/backend/manager/registry.py

```diff
@@ -2363,31 +2370,17 @@ async def _decrease_concurrency_used(access_key: AccessKey, is_private: bool) ->
                     await SessionRow.set_session_status(
                         self.db, session_id, SessionStatus.CANCELLED
                     )
-                case SessionStatus.PULLING:
-                    # Exceptionally allow superadmins to destroy PULLING sessions.
-                    # Clients should be informed that they have to handle the containers destroyed here.
-                    # TODO: detach image-pull process from kernel-start process and allow all users to destroy PULLING sessions.
-                    if forced and user_role == UserRole.SUPERADMIN:
-                        log.warning(
-                            "force-terminating session (s:{}, status:{})",
-                            session_id,
-                            target_session.status,
-                        )
-                        await _force_destroy_for_suadmin(SessionStatus.CANCELLED)
-                        await _decrease_concurrency_used(
-                            target_session.access_key, target_session.is_private
-                        )
-                        return {}
-                    raise GenericForbidden("Cannot destroy sessions in pulling status")
                 case (
```

Comment:
Is it okay if this behavior is removed?

Reply:
yeah, this is one of the purposes we expected from this refactoring.
but I think it would be better to do this work in another branch cause working in fine grained PRs makes it easy to review